### PR TITLE
Add URL resolving for SeleniumProtocol, unify ConfUtils method interfaces

### PR DESCRIPTION
**Pull request by FelixEngl**
Hello @jnioche,

while working with docker and SC I recognized, that the SeleniumProtocol has sometimes problems with resolving the name of the SeleniumHub like `http://seleniumhub:4444` to an IP-address.
To fix that unwanted behaviour of failing to resolve the host, I wrote a backwards-compatible extension for the `selenium.addresses` config part. (Old configs should have no problem with the new extension, you can check that in the unit tests.)

Furthermore I cleaned up the ConfUtils and cleaned the irregular method interfaces (before that it was a mix of `<method>(String, Config)` and `<method>(Config, String)`). I also added some additional helper methods that should improve the ease of use of the config in general.

Best Regards

Felix

Developer Certificate of Origin
===============================

By contributing to StormCrawler, you accept and agree to the following terms and conditions (the *Developer Certificate of Origin*) for your present and future contributions submitted to StormCrawler. 
Please refer to the *Developer Certificate of Origin* section in [`CONTRIBUTING.md`](CONTRIBUTING.md) for details.

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

Before opening a PR, please check that: 

* You've squashed your commits into a single one
* You've described what the PR does or at least point to a related issue
* You've signed-ff your commits with 'git commit -s'
* The code is properly formatted with 'mvn git-code-format:format-code -Dgcf.globPattern=**/*'

Thanks!

### CODE REVIEW

File: core/src/main/java/com/digitalpebble/stormcrawler/protocol/selenium/RemoteDriverProtocol.java [hunk 1]
```diff
@@ -15,25 +15,108 @@
...
+    public static final String SELENIUM_IMPLICIT_WAIT = "selenium.implicitlyWait";
+    public static final String SELENIUM_PAYLOAD_TIMEOUT = "selenium.pageLoadTimeout";
+    public static final String SELENIUM_SCRIPT_TIMEOUT = "selenium.scriptTimeout";
+
+    @Nullable
+    public static List<ResolvedUrl> loadURLsFromConfig(@NotNull Config conf)
+            throws MalformedURLException {
+        Collection<Object> collection = ConfUtils.loadCollectionOrNull(conf, SELENIUM_ADDRESSES);
+        if (collection == null) return null;
+        ArrayList<ResolvedUrl> retVal = new ArrayList<>(collection.size());
```
Comment (rzo1) [inline core/src/main/java/com/digitalpebble/stormcrawler/protocol/selenium/RemoteDriverProtocol.java:83]: Can we use `List` instead?

File: core/src/main/java/com/digitalpebble/stormcrawler/util/URLResolver.java [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Describes the strategies used by the RemoteDriverProtocol to resolve the given addresses. This is
+ * sometimes necessary, when running in a docker container or special environment.
+ */
+// TODO: Right now it only resolves to the first possible target. Maybe we need some kind of
```
Comment (rzo1) [inline core/src/main/java/com/digitalpebble/stormcrawler/util/URLResolver.java:27]: If this is merged, please create an issue to track this TODO ;)

File: core/src/main/java/com/digitalpebble/stormcrawler/util/ConfUtils.java [hunk 3]
```diff
@@ -14,73 +14,240 @@
...
+            throw new RuntimeException(String.format("There is no SubConfig at key '%s'.", key));
+        }
+        return subConfig;
+    }
+
     /**
      * Return one or more Strings regardless of whether they are represented as a single String or a
      * list in the config or an empty List if no value could be found for that key.
      */
-    public static List<String> loadListFromConf(String paramKey, Map stormConf) {
```
Comment (rzo1) [inline core/src/main/java/com/digitalpebble/stormcrawler/util/ConfUtils.java:81]: Might break user code, maybe provide additional `public` method with original signature?

Comment (FelixEngl) [reply core/src/main/java/com/digitalpebble/stormcrawler/util/ConfUtils.java:81]: yeah, i forgot to deprecate it. ^^'
I'll correct it when I get an approval of jnioche that he accepts the presented feature.

**Pull request closed without merging**